pepper-apis/src/main/java/org/broadinstitute/ddp/model/dsm/ParticipantStatusTrackingInfo.java

```diff
@@ -48,70 +49,85 @@ private RecordStatus figureOutMedicalRecordStatus(
         } else if (requested == null && received != null) {
             return RecordStatus.RECEIVED;
         } else {
-            LOG.error("Something completely unexpected happened with ", entityName);
+            LOG.error("Something completely unexpected happened with {}", entityName);
             return RecordStatus.UNKNOWN;
         }
     }
 
-    public ParticipantStatusTrackingInfo(ParticipantStatus dsmParticipantStatus,
-                                         List<Workflow> workflows,
-                                         EnrollmentStatusType enrollmentStatusType,
-                                         String userGuid
-    ) {
-        this(dsmParticipantStatus, enrollmentStatusType, userGuid);
-        this.workflows = workflows;
-    }
-
-    public ParticipantStatusTrackingInfo(
-            ParticipantStatus dsmParticipantStatus,
-            EnrollmentStatusType enrollmentStatusType,
-            String userGuid
-    ) {
+    public ParticipantStatusTrackingInfo(ParticipantStatusES statusES, EnrollmentStatusType enrollmentStatusType,
+                                         String userGuid) {
         this.userGuid = userGuid;
+        this.workflows = statusES.getWorkflows() != null ? statusES.getWorkflows() : new ArrayList<>();
+
+        LocalDate minReceived = null;
+        LocalDate minRequested = null;
+        if (statusES.getMedicalRecords() != null) {
+            for (ParticipantStatusES.MedicalRecord medicalRecord : statusES.getMedicalRecords()) {
+                if (medicalRecord.getReceived() != null) {
+                    if (minReceived == null || minReceived.compareTo(medicalRecord.getReceived()) > 0) {
+                        minReceived = medicalRecord.getReceived();
+                    }
+                }
+                if (medicalRecord.getRequested() != null) {
+                    if (minRequested == null || minRequested.compareTo(medicalRecord.getRequested()) > 0) {
```

Comment:
We always return a medical or tissue record object, even if there's no timestamps. Before, DSM will compute the timestamps to use. Now, we're querying the data directly from ES, so we have to do this computation of finding the minimum timestamp to use.

Reply:
Variable names something like latestReceivedDate , latestRequestedDate will be more clear !

Reply:
Actually, we're looking for the minimum date, so the oldest date and not latest.